### PR TITLE
[WIP] Add kubetest "git" extract strategy

### PR DESCRIPTION
**Pull request by clarketm**
Add a new *extract* strategy to `kubetest` to get kube git version. (e.g. `--extract=git`)

/assign @krzyzacy

### CODE REVIEW

File: kubetest/extract_k8s.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ type extractMode int
 const (
 	none    extractMode = iota
 	local               // local
+	git                 // A git tag version from the repository (e.g. v1.13.7-gke.29)
```
Comment (BenTheElder) [inline kubetest/extract_k8s.go]: is this only a local clone then? it also doesn't seem to fetch tags etc.?

why would we use this over --local and doing a `git clone <k8s-fork> && git checkout vfoo` first?

Comment (clarketm) [reply kubetest/extract_k8s.go]: The idea is to leverage the *version* from git tag, using the same method as `hack/print-workspace-status.sh`, and use that to determine the built kubernetes version to `extract`.

**Comment by clarketm** [comment]
Obsolete.

**Comment by clarketm** [comment]
Replaced by #14823
/close

**Pull request closed without merging**